src/openloco/windows/toolbar_top.cpp

```diff
@@ -0,0 +1,1236 @@
+
+        build_vehicle::open(itemIndex, 1 << 31);
+    }
+
+    // 0x0043ABCB
+    static void vehicles_menu_mouse_down(window* window, widget_index widgetIndex)
+    {
+        auto player_company_id = companymgr::get_controlling_id();
+        auto company = companymgr::get(player_company_id);
+        uint16_t available_vehicles = company->available_vehicles;
+
+        auto company_colour = (1 << 29) | (companymgr::get_player_company_colour() << 19);
+        auto interface = objectmgr::get<interface_skin_object>();
+
+        uint16_t vehicle_counts[thingmgr::num_thing_lists]{ 0 };
+        for (openloco::vehicle* v = thingmgr::first<openloco::vehicle>(); v != nullptr; v = v->next_vehicle())
+        {
+            if (v->owner != player_company_id)
+                continue;
+
+            if ((v->var_38 & (1 << 4)) != 0)
+                continue;
+
+            vehicle_counts[v->var_5E]++;
+        }
+
+        uint8_t ddIndex = 0;
+        for (uint16_t vehicleIdx = 0; vehicleIdx < thingmgr::num_thing_lists; vehicleIdx++)
+        {
+            if ((available_vehicles & (1 << vehicleIdx)) == 0)
+                continue;
+
+            static const uint32_t vehicle_images[] = {
+                interface_skin::image_ids::vehicle_train,
+                interface_skin::image_ids::vehicle_bus,
+                interface_skin::image_ids::vehicle_truck,
+                interface_skin::image_ids::vehicle_tram,
+                interface_skin::image_ids::vehicle_aircraft,
+                interface_skin::image_ids::vehicle_ship,
+            };
+
+            uint32_t vehicle_image = company_colour | vehicle_images[vehicleIdx];
+
+            static const string_id num_singular[] = {
+                string_ids::num_trains_singular,
+                string_ids::num_buses_singular,
+                string_ids::num_trucks_singular,
+                string_ids::num_trams_singular,
+                string_ids::num_aircrafts_singular,
+                string_ids::num_ships_singular,
+            };
+
+            static const string_id num_plural[] = {
+                string_ids::num_trains_plural,
+                string_ids::num_buses_plural,
+                string_ids::num_trucks_plural,
+                string_ids::num_trams_plural,
+                string_ids::num_aircraft_plural,
+                string_ids::num_ships_plural,
+            };
+
+            uint16_t vehicle_count = vehicle_counts[vehicleIdx];
+
+            string_id vehicle_string_id;
+            if (vehicle_count == 1)
+                vehicle_string_id = num_singular[vehicleIdx];
+            else
+                vehicle_string_id = num_plural[vehicleIdx];
+
+            dropdown::add(ddIndex, string_ids::menu_sprite_stringid, { interface->img + vehicle_image, vehicle_string_id, vehicle_count });
+            menu_options[ddIndex] = vehicleIdx;
+            ddIndex++;
+        }
+
+        dropdown::show_below(window, widgetIndex, ddIndex, 25);
```

Comment:
This is missing a `dropdown::set_highlighted_item` call involving `last_vehicles_option`